### PR TITLE
[a.i/ii/iv/v] Code factoring on handlers / routes

### PR DESCRIPTION
**Pull request by b-z**

### CODE REVIEW

**Comment by b-z** [comment]
After fixing the login module, we should remove the temporary checkLogin_ function, and fix some TODOs in handlers.go

File: server/handlers.go [hunk 1]
```diff
@@ -184,38 +181,25 @@ func cancelProductHandler(c *gin.Context, client *mongo.Client) {
 	claims, _ := checkLogin(c)
 	userID := claims.Username
 
-	if data.Cancel1(client, string(userID), string(id)) != nil {
+	if data.Cancel1(client, string(userID), string(id)) != nil { // TODO: rename function here
```
Comment (VamshiTeja) [inline server/handlers.go]: needs to change this to data.CancelOrder() functions including whoCancelled bool argument

File: server/handlers.go [hunk 2]
```diff
@@ -184,38 +181,25 @@ func cancelProductHandler(c *gin.Context, client *mongo.Client) {
...
 		c.String(500, "Cancelation failed.")
 		return
 	}
 
-	if data.Cancel2(client, string(userID), string(id)) != nil {
+	if data.Cancel2(client, string(userID), string(id)) != nil { // TODO: rename function here
```
Comment (VamshiTeja) [inline server/handlers.go]: same here as above

File: server/router.go [hunk 3]
```diff
@@ -6,31 +6,30 @@ import (
...
+	dbClient := Connect2DB()
 	router.GET("/", hello)
-	router.GET("/login", loginPage)
-	router.GET("/list", attachDB(client, listPage))
+	router.GET("/login", loginPage)                   // TODO: not used
+	router.GET("/list", attachDB(dbClient, listPage)) // TODO: not used
 
-	router.GET("/products", attachDB(client, getAllProductsHandler))
-	router.GET("/products/:id", attachDB(client, getProductByIDHandler))
-	router.GET("/products-user/:id", attachDB(client, getProductByUserIDHandler))
```
Comment (VamshiTeja) [inline server/router.go:17]: dont we need this?

File: server/router.go [hunk 4]
```diff
@@ -6,31 +6,30 @@ import (
...
-	router.GET("/users", attachDB(client, GetUserByCookie))
+	router.GET("/users/:id/products", attachDB(dbClient, getProductByUserIDHandler))
+	router.GET("/users/:id/orders", attachDB(dbClient, getOrderByUserIDHandler))
+	router.GET("/users/:id", attachDB(dbClient, getUserByIDHandler))
+
+	router.POST("/login", attachDB(dbClient, signIn))
+	router.GET("/welcome", welcome)
+	router.POST("/refresh", refresh)
+	router.POST("/signup", attachDB(dbClient, signUp))
+	router.GET("/cookie", attachDB(dbClient, getUserByCookie))
```
Comment (VamshiTeja) [inline server/router.go]: may be "/usercookie" what do you think?

**Review by VamshiTeja — CHANGES_REQUESTED** [review]
Please Change cancel1 and cancel2 functions to CancelOrder wherever necessary

File: server/router.go (hunk 3 above)
Comment (b-z) [reply server/router.go:17]: we need, and I have change the url of them,
for example, the third one -> /users/:id/products

File: server/router.go (hunk 4 above)
Comment (b-z) [reply server/router.go]: ok

File: server/router.go (hunk 3 above)
Comment (VamshiTeja) [reply server/router.go:17]: cool

**Comment by b-z** [comment]
fixed

**Review by VamshiTeja — APPROVED** [review]

**Pull request merged**